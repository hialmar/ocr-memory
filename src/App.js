import React, { Component } from 'react'
import shuffle from 'lodash.shuffle'

import './App.css'

import Card from './Card'
import GuessCount from './GuessCount'
import HallOfFame, { FAKE_HOF } from './HallOfFame'

const SIDE = 6
const SYMBOLS = '😀🎉💖🎩🐶🐱🦄🐬🌍🌛🌞💫🍎🍌🍓🍐🍟🍿'

class App extends Component {
  cards = this.generateCards()

  /* // première approche pour préserver le this
  constructor(props) {
    super(props)
    this.handleCardClick = this.handleCardClick.bind(this);
  }
  */

  generateCards() {
    const result = []
    const size = SIDE * SIDE
    const candidates = shuffle(SYMBOLS)
    while (result.length < size) {
      const card = candidates.pop()
      result.push(card, card)
    }
    return shuffle(result)
  }

  /* // premiere approche pour préserver le this
  handleCardClick(card) {
    console.log(card, 'clicked', this)
  }
   */

  // seconde approche pour préserver le this
  // arrow fct for binding
  handleCardClick = card => {
    console.log(card, 'clicked', this)
  }



  // troisième approche pour préserver le this
  // pas encore géré par babel
  /*
  @autobind
  handleCardClick(card) {
    console.log(card, 'clicked', this)
  }
   */

  render() {
    const won = new Date().getSeconds() % 2 === 0
    return (
      <div className="memory">
        <GuessCount />
        {this.cards.map((card, index) => (
          <Card card={card} feedback="visible" key={index} onClick={this.handleCardClick} />
        ))}
        {won && <HallOfFame entries={FAKE_HOF} />}
      </div>
    )
  }
}

export default App
