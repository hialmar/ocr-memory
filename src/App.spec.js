import { expect } from 'chai'
import React from 'react'
import { shallow } from 'enzyme'

import App, { SYMBOLS } from './App'

import GuessCount from './GuessCount'
import sinon from 'sinon'
import Card from './Card'

describe('<App />', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<App />)

    expect(wrapper).to.contain(<GuessCount guesses={0}/>)
  })

  it('has 36 cards', () => {
    const wrapper = shallow(<App />)

    expect(wrapper.find('Card')).to.have.length(36)
    })

  it('should match its reference snapshot', () => {
    const mock = sinon
      .stub(App.prototype, 'generateCards')
      .returns([...SYMBOLS.repeat(2)])
    try {
      const wrapper = shallow(<App />)

      let snapshotFilename = __filename + ".snap";
      let snapshotName = "should match its reference snapshot";
      expect(wrapper).to.matchSnapshot(snapshotFilename, snapshotName);
      //expect(wrapper).to.matchSnapshot()
    } finally {
      mock.restore()
    }
  })
})



/*
it('demoes', () => {
  expect({ name: 'Joe'}).toEqual({name: 'Jane'})
})
*/
