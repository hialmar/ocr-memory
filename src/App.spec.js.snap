// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match its reference snapshot`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <GuessCount
          guesses={0}
        />,
        Array [
          <Card
            card="üòÄ"
            feedback="hidden"
            index={0}
            onClick={[Function]}
          />,
          <Card
            card="üéâ"
            feedback="hidden"
            index={1}
            onClick={[Function]}
          />,
          <Card
            card="üíñ"
            feedback="hidden"
            index={2}
            onClick={[Function]}
          />,
          <Card
            card="üé©"
            feedback="hidden"
            index={3}
            onClick={[Function]}
          />,
          <Card
            card="üê∂"
            feedback="hidden"
            index={4}
            onClick={[Function]}
          />,
          <Card
            card="üê±"
            feedback="hidden"
            index={5}
            onClick={[Function]}
          />,
          <Card
            card="ü¶Ñ"
            feedback="hidden"
            index={6}
            onClick={[Function]}
          />,
          <Card
            card="üê¨"
            feedback="hidden"
            index={7}
            onClick={[Function]}
          />,
          <Card
            card="üåç"
            feedback="hidden"
            index={8}
            onClick={[Function]}
          />,
          <Card
            card="üåõ"
            feedback="hidden"
            index={9}
            onClick={[Function]}
          />,
          <Card
            card="üåû"
            feedback="hidden"
            index={10}
            onClick={[Function]}
          />,
          <Card
            card="üí´"
            feedback="hidden"
            index={11}
            onClick={[Function]}
          />,
          <Card
            card="üçé"
            feedback="hidden"
            index={12}
            onClick={[Function]}
          />,
          <Card
            card="üçå"
            feedback="hidden"
            index={13}
            onClick={[Function]}
          />,
          <Card
            card="üçì"
            feedback="hidden"
            index={14}
            onClick={[Function]}
          />,
          <Card
            card="üçê"
            feedback="hidden"
            index={15}
            onClick={[Function]}
          />,
          <Card
            card="üçü"
            feedback="hidden"
            index={16}
            onClick={[Function]}
          />,
          <Card
            card="üçø"
            feedback="hidden"
            index={17}
            onClick={[Function]}
          />,
          <Card
            card="üòÄ"
            feedback="hidden"
            index={18}
            onClick={[Function]}
          />,
          <Card
            card="üéâ"
            feedback="hidden"
            index={19}
            onClick={[Function]}
          />,
          <Card
            card="üíñ"
            feedback="hidden"
            index={20}
            onClick={[Function]}
          />,
          <Card
            card="üé©"
            feedback="hidden"
            index={21}
            onClick={[Function]}
          />,
          <Card
            card="üê∂"
            feedback="hidden"
            index={22}
            onClick={[Function]}
          />,
          <Card
            card="üê±"
            feedback="hidden"
            index={23}
            onClick={[Function]}
          />,
          <Card
            card="ü¶Ñ"
            feedback="hidden"
            index={24}
            onClick={[Function]}
          />,
          <Card
            card="üê¨"
            feedback="hidden"
            index={25}
            onClick={[Function]}
          />,
          <Card
            card="üåç"
            feedback="hidden"
            index={26}
            onClick={[Function]}
          />,
          <Card
            card="üåõ"
            feedback="hidden"
            index={27}
            onClick={[Function]}
          />,
          <Card
            card="üåû"
            feedback="hidden"
            index={28}
            onClick={[Function]}
          />,
          <Card
            card="üí´"
            feedback="hidden"
            index={29}
            onClick={[Function]}
          />,
          <Card
            card="üçé"
            feedback="hidden"
            index={30}
            onClick={[Function]}
          />,
          <Card
            card="üçå"
            feedback="hidden"
            index={31}
            onClick={[Function]}
          />,
          <Card
            card="üçì"
            feedback="hidden"
            index={32}
            onClick={[Function]}
          />,
          <Card
            card="üçê"
            feedback="hidden"
            index={33}
            onClick={[Function]}
          />,
          <Card
            card="üçü"
            feedback="hidden"
            index={34}
            onClick={[Function]}
          />,
          <Card
            card="üçø"
            feedback="hidden"
            index={35}
            onClick={[Function]}
          />,
        ],
        false,
      ],
      "className": "memory",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "guesses": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "card": "üòÄ",
          "feedback": "hidden",
          "index": 0,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "card": "üéâ",
          "feedback": "hidden",
          "index": 1,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "card": "üíñ",
          "feedback": "hidden",
          "index": 2,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "card": "üé©",
          "feedback": "hidden",
          "index": 3,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "function",
        "props": Object {
          "card": "üê∂",
          "feedback": "hidden",
          "index": 4,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "function",
        "props": Object {
          "card": "üê±",
          "feedback": "hidden",
          "index": 5,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "6",
        "nodeType": "function",
        "props": Object {
          "card": "ü¶Ñ",
          "feedback": "hidden",
          "index": 6,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "7",
        "nodeType": "function",
        "props": Object {
          "card": "üê¨",
          "feedback": "hidden",
          "index": 7,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "8",
        "nodeType": "function",
        "props": Object {
          "card": "üåç",
          "feedback": "hidden",
          "index": 8,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "9",
        "nodeType": "function",
        "props": Object {
          "card": "üåõ",
          "feedback": "hidden",
          "index": 9,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "10",
        "nodeType": "function",
        "props": Object {
          "card": "üåû",
          "feedback": "hidden",
          "index": 10,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "11",
        "nodeType": "function",
        "props": Object {
          "card": "üí´",
          "feedback": "hidden",
          "index": 11,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "12",
        "nodeType": "function",
        "props": Object {
          "card": "üçé",
          "feedback": "hidden",
          "index": 12,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "13",
        "nodeType": "function",
        "props": Object {
          "card": "üçå",
          "feedback": "hidden",
          "index": 13,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "14",
        "nodeType": "function",
        "props": Object {
          "card": "üçì",
          "feedback": "hidden",
          "index": 14,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "15",
        "nodeType": "function",
        "props": Object {
          "card": "üçê",
          "feedback": "hidden",
          "index": 15,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "16",
        "nodeType": "function",
        "props": Object {
          "card": "üçü",
          "feedback": "hidden",
          "index": 16,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "17",
        "nodeType": "function",
        "props": Object {
          "card": "üçø",
          "feedback": "hidden",
          "index": 17,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "18",
        "nodeType": "function",
        "props": Object {
          "card": "üòÄ",
          "feedback": "hidden",
          "index": 18,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "19",
        "nodeType": "function",
        "props": Object {
          "card": "üéâ",
          "feedback": "hidden",
          "index": 19,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "20",
        "nodeType": "function",
        "props": Object {
          "card": "üíñ",
          "feedback": "hidden",
          "index": 20,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "21",
        "nodeType": "function",
        "props": Object {
          "card": "üé©",
          "feedback": "hidden",
          "index": 21,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "22",
        "nodeType": "function",
        "props": Object {
          "card": "üê∂",
          "feedback": "hidden",
          "index": 22,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "23",
        "nodeType": "function",
        "props": Object {
          "card": "üê±",
          "feedback": "hidden",
          "index": 23,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "24",
        "nodeType": "function",
        "props": Object {
          "card": "ü¶Ñ",
          "feedback": "hidden",
          "index": 24,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "25",
        "nodeType": "function",
        "props": Object {
          "card": "üê¨",
          "feedback": "hidden",
          "index": 25,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "26",
        "nodeType": "function",
        "props": Object {
          "card": "üåç",
          "feedback": "hidden",
          "index": 26,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "27",
        "nodeType": "function",
        "props": Object {
          "card": "üåõ",
          "feedback": "hidden",
          "index": 27,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "28",
        "nodeType": "function",
        "props": Object {
          "card": "üåû",
          "feedback": "hidden",
          "index": 28,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "29",
        "nodeType": "function",
        "props": Object {
          "card": "üí´",
          "feedback": "hidden",
          "index": 29,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "30",
        "nodeType": "function",
        "props": Object {
          "card": "üçé",
          "feedback": "hidden",
          "index": 30,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "31",
        "nodeType": "function",
        "props": Object {
          "card": "üçå",
          "feedback": "hidden",
          "index": 31,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "32",
        "nodeType": "function",
        "props": Object {
          "card": "üçì",
          "feedback": "hidden",
          "index": 32,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "33",
        "nodeType": "function",
        "props": Object {
          "card": "üçê",
          "feedback": "hidden",
          "index": 33,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "34",
        "nodeType": "function",
        "props": Object {
          "card": "üçü",
          "feedback": "hidden",
          "index": 34,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "35",
        "nodeType": "function",
        "props": Object {
          "card": "üçø",
          "feedback": "hidden",
          "index": 35,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <GuessCount
            guesses={0}
          />,
          Array [
            <Card
              card="üòÄ"
              feedback="hidden"
              index={0}
              onClick={[Function]}
            />,
            <Card
              card="üéâ"
              feedback="hidden"
              index={1}
              onClick={[Function]}
            />,
            <Card
              card="üíñ"
              feedback="hidden"
              index={2}
              onClick={[Function]}
            />,
            <Card
              card="üé©"
              feedback="hidden"
              index={3}
              onClick={[Function]}
            />,
            <Card
              card="üê∂"
              feedback="hidden"
              index={4}
              onClick={[Function]}
            />,
            <Card
              card="üê±"
              feedback="hidden"
              index={5}
              onClick={[Function]}
            />,
            <Card
              card="ü¶Ñ"
              feedback="hidden"
              index={6}
              onClick={[Function]}
            />,
            <Card
              card="üê¨"
              feedback="hidden"
              index={7}
              onClick={[Function]}
            />,
            <Card
              card="üåç"
              feedback="hidden"
              index={8}
              onClick={[Function]}
            />,
            <Card
              card="üåõ"
              feedback="hidden"
              index={9}
              onClick={[Function]}
            />,
            <Card
              card="üåû"
              feedback="hidden"
              index={10}
              onClick={[Function]}
            />,
            <Card
              card="üí´"
              feedback="hidden"
              index={11}
              onClick={[Function]}
            />,
            <Card
              card="üçé"
              feedback="hidden"
              index={12}
              onClick={[Function]}
            />,
            <Card
              card="üçå"
              feedback="hidden"
              index={13}
              onClick={[Function]}
            />,
            <Card
              card="üçì"
              feedback="hidden"
              index={14}
              onClick={[Function]}
            />,
            <Card
              card="üçê"
              feedback="hidden"
              index={15}
              onClick={[Function]}
            />,
            <Card
              card="üçü"
              feedback="hidden"
              index={16}
              onClick={[Function]}
            />,
            <Card
              card="üçø"
              feedback="hidden"
              index={17}
              onClick={[Function]}
            />,
            <Card
              card="üòÄ"
              feedback="hidden"
              index={18}
              onClick={[Function]}
            />,
            <Card
              card="üéâ"
              feedback="hidden"
              index={19}
              onClick={[Function]}
            />,
            <Card
              card="üíñ"
              feedback="hidden"
              index={20}
              onClick={[Function]}
            />,
            <Card
              card="üé©"
              feedback="hidden"
              index={21}
              onClick={[Function]}
            />,
            <Card
              card="üê∂"
              feedback="hidden"
              index={22}
              onClick={[Function]}
            />,
            <Card
              card="üê±"
              feedback="hidden"
              index={23}
              onClick={[Function]}
            />,
            <Card
              card="ü¶Ñ"
              feedback="hidden"
              index={24}
              onClick={[Function]}
            />,
            <Card
              card="üê¨"
              feedback="hidden"
              index={25}
              onClick={[Function]}
            />,
            <Card
              card="üåç"
              feedback="hidden"
              index={26}
              onClick={[Function]}
            />,
            <Card
              card="üåõ"
              feedback="hidden"
              index={27}
              onClick={[Function]}
            />,
            <Card
              card="üåû"
              feedback="hidden"
              index={28}
              onClick={[Function]}
            />,
            <Card
              card="üí´"
              feedback="hidden"
              index={29}
              onClick={[Function]}
            />,
            <Card
              card="üçé"
              feedback="hidden"
              index={30}
              onClick={[Function]}
            />,
            <Card
              card="üçå"
              feedback="hidden"
              index={31}
              onClick={[Function]}
            />,
            <Card
              card="üçì"
              feedback="hidden"
              index={32}
              onClick={[Function]}
            />,
            <Card
              card="üçê"
              feedback="hidden"
              index={33}
              onClick={[Function]}
            />,
            <Card
              card="üçü"
              feedback="hidden"
              index={34}
              onClick={[Function]}
            />,
            <Card
              card="üçø"
              feedback="hidden"
              index={35}
              onClick={[Function]}
            />,
          ],
          false,
        ],
        "className": "memory",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "guesses": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "card": "üòÄ",
            "feedback": "hidden",
            "index": 0,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "card": "üéâ",
            "feedback": "hidden",
            "index": 1,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "card": "üíñ",
            "feedback": "hidden",
            "index": 2,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "card": "üé©",
            "feedback": "hidden",
            "index": 3,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "card": "üê∂",
            "feedback": "hidden",
            "index": 4,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "function",
          "props": Object {
            "card": "üê±",
            "feedback": "hidden",
            "index": 5,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "function",
          "props": Object {
            "card": "ü¶Ñ",
            "feedback": "hidden",
            "index": 6,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "7",
          "nodeType": "function",
          "props": Object {
            "card": "üê¨",
            "feedback": "hidden",
            "index": 7,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "8",
          "nodeType": "function",
          "props": Object {
            "card": "üåç",
            "feedback": "hidden",
            "index": 8,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "9",
          "nodeType": "function",
          "props": Object {
            "card": "üåõ",
            "feedback": "hidden",
            "index": 9,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "10",
          "nodeType": "function",
          "props": Object {
            "card": "üåû",
            "feedback": "hidden",
            "index": 10,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "11",
          "nodeType": "function",
          "props": Object {
            "card": "üí´",
            "feedback": "hidden",
            "index": 11,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "12",
          "nodeType": "function",
          "props": Object {
            "card": "üçé",
            "feedback": "hidden",
            "index": 12,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "13",
          "nodeType": "function",
          "props": Object {
            "card": "üçå",
            "feedback": "hidden",
            "index": 13,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "14",
          "nodeType": "function",
          "props": Object {
            "card": "üçì",
            "feedback": "hidden",
            "index": 14,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "15",
          "nodeType": "function",
          "props": Object {
            "card": "üçê",
            "feedback": "hidden",
            "index": 15,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "16",
          "nodeType": "function",
          "props": Object {
            "card": "üçü",
            "feedback": "hidden",
            "index": 16,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "17",
          "nodeType": "function",
          "props": Object {
            "card": "üçø",
            "feedback": "hidden",
            "index": 17,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "18",
          "nodeType": "function",
          "props": Object {
            "card": "üòÄ",
            "feedback": "hidden",
            "index": 18,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "19",
          "nodeType": "function",
          "props": Object {
            "card": "üéâ",
            "feedback": "hidden",
            "index": 19,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "20",
          "nodeType": "function",
          "props": Object {
            "card": "üíñ",
            "feedback": "hidden",
            "index": 20,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "21",
          "nodeType": "function",
          "props": Object {
            "card": "üé©",
            "feedback": "hidden",
            "index": 21,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "22",
          "nodeType": "function",
          "props": Object {
            "card": "üê∂",
            "feedback": "hidden",
            "index": 22,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "23",
          "nodeType": "function",
          "props": Object {
            "card": "üê±",
            "feedback": "hidden",
            "index": 23,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "24",
          "nodeType": "function",
          "props": Object {
            "card": "ü¶Ñ",
            "feedback": "hidden",
            "index": 24,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "25",
          "nodeType": "function",
          "props": Object {
            "card": "üê¨",
            "feedback": "hidden",
            "index": 25,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "26",
          "nodeType": "function",
          "props": Object {
            "card": "üåç",
            "feedback": "hidden",
            "index": 26,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "27",
          "nodeType": "function",
          "props": Object {
            "card": "üåõ",
            "feedback": "hidden",
            "index": 27,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "28",
          "nodeType": "function",
          "props": Object {
            "card": "üåû",
            "feedback": "hidden",
            "index": 28,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "29",
          "nodeType": "function",
          "props": Object {
            "card": "üí´",
            "feedback": "hidden",
            "index": 29,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "30",
          "nodeType": "function",
          "props": Object {
            "card": "üçé",
            "feedback": "hidden",
            "index": 30,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "31",
          "nodeType": "function",
          "props": Object {
            "card": "üçå",
            "feedback": "hidden",
            "index": 31,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "32",
          "nodeType": "function",
          "props": Object {
            "card": "üçì",
            "feedback": "hidden",
            "index": 32,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "33",
          "nodeType": "function",
          "props": Object {
            "card": "üçê",
            "feedback": "hidden",
            "index": 33,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "34",
          "nodeType": "function",
          "props": Object {
            "card": "üçü",
            "feedback": "hidden",
            "index": 34,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "35",
          "nodeType": "function",
          "props": Object {
            "card": "üçø",
            "feedback": "hidden",
            "index": 35,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
